inst/templates/ad_adsl.R

```diff
@@ -194,29 +194,49 @@ adsl <- adsl %>%
 ## Last known alive date ----
 ## DTC variables are converted to numeric dates imputing missing day and month
 ## to the first
-ae_start_date <- date_source(
-  dataset_name = "ae",
-  date = convert_dtc_to_dt(AESTDTC, highest_imputation = "M")
-)
-ae_end_date <- date_source(
-  dataset_name = "ae",
-  date = convert_dtc_to_dt(AEENDTC, highest_imputation = "M")
-)
-lb_date <- date_source(
-  dataset_name = "lb",
-  date = convert_dtc_to_dt(LBDTC, highest_imputation = "M")
-)
-trt_end_date <- date_source(
-  dataset_name = "adsl",
-  date = TRTEDT
-)
 
 adsl <- adsl %>%
-  derive_var_extreme_dt(
-    new_var = LSTALVDT,
-    ae_start_date, ae_end_date, lb_date, trt_end_date,
+  derive_vars_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    events = list(
+      event(
+        dataset_name = "ae",
+        order = exprs(AESTDTC, AESEQ),
+        condition = !is.na(AESTDTC),
+        set_values_to = exprs(
+          LSTALVDT = convert_dtc_to_dt(AESTDTC, highest_imputation = "M"),
+          seq = AESEQ
+        ),
+      ),
+      event(
+        dataset_name = "ae",
+        order = exprs(AEENDTC, AESEQ),
+        condition = !is.na(AEENDTC),
+        set_values_to = exprs(
+          LSTALVDT = convert_dtc_to_dt(AEENDTC, highest_imputation = "M"),
+          seq = AESEQ
+        ),
+      ),
+      event(
+        dataset_name = "lb",
+        order = exprs(LBDTC, LBSEQ),
+        condition = !is.na(LBDTC),
+        set_values_to = exprs(
+          LSTALVDT = convert_dtc_to_dt(LBDTC, highest_imputation = "M"),
+          seq = LBSEQ
+        ),
+      ),
+      event(
+        dataset_name = "adsl",
+        condition = !is.na(TRTEDT),
+        set_values_to = exprs(LSTALVDT = TRTEDT, seq = NA_integer_),
+      )
+    ),
     source_datasets = list(ae = ae, lb = lb, adsl = adsl),
-    mode = "last"
+    tmp_event_nr_var = event_nr,
+    order = exprs(LSTALVDT, seq, event_nr),
+    mode = "last",
+    new_vars = exprs(LSTALVDT = LSTALVDT)
```

Comment:
Can be simplified to `new_vars = exprs(LSTALVDT)`.

Reply:
Updated, thanks!